### DISCUSSION TITLE
Lesson - 7 why is testOwnerIsMsgSender == i_owner is msg.sender ?

### DISCUSSION THREAD

**Discussion by SquilliamX** [Cyfrin/foundry-full-course-cu#640] [Q&A]
I'm curious about the positive test result for this function:

'''
function testOwnerIsMsgSender() public{
    assertEq(fundMe.i_owner(), msg.sender);
}
'''

This might be due to the fact that the FundMeTest contract generates DeployFundMe, which in turn creates FundMe. As a result, the i_owner of FundMe could indeed be set to the address of DeployFundMe, rather than our own (msg.sender) address. Am I correct?

**Comment by theirrationalone** [comment]
Yes It's bare True....

You can use `vm.prank` or `vm.startPrank` **cheatcodes** to set a specific owner.

and Please don't wane your patience follow Patrick's Video thoroughly, He described everything Perfectly!

**Comment by shikhar229169** [comment]
Hey, check this out:
https://github.com/Cyfrin/foundry-full-course-f23/discussions/504
https://github.com/Cyfrin/foundry-full-course-f23/discussions/513
